### PR TITLE
Add spec for `Converters::CSV`

### PR DESCRIPTION
**Pull request by AI-Mozi**
#1

### CODE REVIEW

File: spec/converters/csv_spec.rb [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+RSpec.describe Ronin::Masscan::Converters::CSV do
+  let(:fixtures_path) { File.expand_path(File.join(__dir__, '..', 'fixtures')) }
+  let(:masscan_path)  { File.join(fixtures_path, 'converter', 'masscan.json') }
+  let(:masscan_file)  { Masscan::OutputFile.new(masscan_path) }
+  let(:ip_addr)       { IPAddr.new('93.184.216.34/32') }
+  let(:timestamp)     { 1629960621 }
+  let(:header)        { Ronin::Masscan::Converters::CSV::HEADER }
+
+  describe '.convert' do
+    let(:expected_row) { "status,open,tcp,80,\"syn,ack\",54,93.184.216.34,#{Time.at(timestamp)}" }
```
Comment (postmodern) [inline spec/converters/csv_spec.rb:14]: Should test converting multiple values into multiple rows.

File: spec/converters/csv_spec.rb [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+
+    it 'must convert masscan file to csv and write it into output' do
+      Tempfile.create do |output_file|
+        subject.convert(masscan_file, output_file)
+        output_file.rewind
+
+        expect(output_file.gets.chomp).to eq(header.join(','))
```
Comment (postmodern) [inline spec/converters/csv_spec.rb:21]: I'd call `.gets` once and save it to a local variable. Also you can use `.gets(chomp: true)`.

File: spec/converters/csv_spec.rb [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+  describe '.masscan_file_to_csv' do
+    let(:expected_row) { "status,open,tcp,80,\"syn,ack\",54,93.184.216.34,#{Time.at(timestamp)}" }
+
+    it 'must convert masscan file to csv and write it into output' do
+      Tempfile.create do |output_file|
+        subject.masscan_file_to_csv(masscan_file, output_file)
+        output_file.rewind
+
+        expect(output_file.gets.chomp).to eq(header.join(','))
+        expect(output_file.gets.chomp).to eq(expected_row)
```
Comment (postmodern) [inline spec/converters/csv_spec.rb:36]: Multiple `.gets` calls.

File: spec/converters/csv_spec.rb [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+      end
+    end
+  end
+
+  describe '.masscan_file_to_csv' do
+    let(:expected_row) { "status,open,tcp,80,\"syn,ack\",54,93.184.216.34,#{Time.at(timestamp)}" }
```
Comment (postmodern) [inline spec/converters/csv_spec.rb:28]: Should test converting at least two rows.

File: spec/fixtures/converter/masscan.json [hunk 5]
```diff
@@ -0,0 +1,3 @@
+[
+  { "ip": "93.184.216.34", "timestamp": "1629960621", "ports": [ {"port": 80, "proto": "tcp", "status": "open", "reason": "syn-ack", "ttl": 54} ] }
+]
```
Comment (postmodern) [inline spec/fixtures/converter/masscan.json:3]: There should be at least two values in the file to properly exercise the code.

File: spec/converters/csv_spec.rb [hunk 6]
```diff
@@ -0,0 +1,152 @@
...
+        expect(output_file.gets.chomp).to eq(header.join(','))
+        expect(output_file.gets.chomp).to eq(expected_row)
+      end
+    end
+  end
+
+  describe '.masscan_file_to_rows' do
+    let(:expected_row) do
+      ['status', :open, :tcp, 80, "syn,ack", 54, ip_addr, Time.at(timestamp)]
+    end
```
Comment (postmodern) [inline spec/converters/csv_spec.rb:44]: Needs multiple rows.

**Review by postmodern — CHANGES_REQUESTED** [review]
Minor comments on the specs. There should be multiple values in the fixtures file so we can test whether the `Converters` actually convert all values from the value, not just the first value.

File: spec/converters/csv_spec.rb (hunk 2 above)
Comment (AI-Mozi) [reply spec/converters/csv_spec.rb]: Didn't `.gets` gets only one line? I think we should test against first 3 lines.

Comment (AI-Mozi) [reply spec/converters/csv_spec.rb]: I used `.readlines(chomp: true)` to save it into variable.

File: spec/fixtures/converters/two_records.json [hunk 7]
```diff
@@ -0,0 +1,5 @@
+[
+{ "ip": "93.184.216.34", "timestamp": "1629960621", "ports": [ {"port": 80, "proto": "tcp", "status": "open", "reason": "syn-ack", "ttl": 54} ] }
+,
+{ "ip": "93.184.216.34", "timestamp": "1629960621", "ports": [ {"port": 80, "proto": "tcp", "service": {"name": "title", "banner": "404 - Not Found"} } ] }
+]
```
Comment (postmodern) [inline spec/fixtures/converters/two_records.json:5]: Maybe rename the file to `input.json`. `two_records.json` seems a bit too specific?

File: spec/converters/csv_spec.rb [hunk 8]
```diff
@@ -0,0 +1,164 @@
...
+  end
+
+  describe '.masscan_file_to_csv' do
+    let(:expected_status_row) { "status,open,tcp,80,\"syn,ack\",54,93.184.216.34,#{timestamp}" }
+    let(:expected_banner_row) { "banner,,,,,,,,tcp,80,93.184.216.34,#{timestamp},html_title,404 - Not Found" }
+
+    it 'must convert masscan file to csv and write it into output' do
+      tempfile = Tempfile.new(['ronin-masscan', '.json'])
+
+      subject.convert(masscan_file, tempfile)
```
Comment (postmodern) [inline spec/converters/csv_spec.rb:38]: Should be `.masscan_file_to_csv`.

**Review by postmodern — CHANGES_REQUESTED** [review]
Spotted one copy/paste error.

**Review by postmodern — APPROVED** [review]

**Pull request merged**